lib/deimos/active_record_consume/batch_consumption.rb

```diff
@@ -155,8 +164,14 @@ def update_database(messages)
       # @return [void]
       def upsert_records(messages)
         record_list = build_records(messages)
-        record_list.filter!(self.method(:should_consume?).to_proc)
+        invalid = filter_records(record_list)
+        if invalid.any?
+          ActiveSupport::Notifications.instrument('batch_consumption.invalid_records', {
```

Comment:
So this doesn't actually move the processing out of the transaction. I'm honestly still a little confused about how important this is. If it's taking a really long time to log invalid messages, you probably shouldn't be doing it? Or we should be looking into why it takes so long?

Reply:
Its important if we want to log datadog metrics for failed records. As of now its currently logging within the transaction but it would be nice to have this live outside the transaction if possible.

I can discuss with the team the current usefulness of the metrics outside of just logging the results